File: frontend/src/interfaces/Mealplan.ts
/* import type { DiseaseInterface } from './Disease';
import type{ MealdayInterface } from './Mealday'; */
export interface MealplanInterface {
    ID?: number;
    PlanName?: string;
    AdminID?: number; //FK
    /* DiseaseID?: number; //FK */
    /* Disease?: DiseaseInterface; */
    /* Mealdays?: MealdayInterface[];    */           
}

File: backend/entity/mealplan.go
package entity

import (
	"gorm.io/gorm"
)

type Mealplan struct {
	gorm.Model
	PlanName string

	/* // AdminID ทำหน้าที่เป็น FK
	AdminID uint
	Admin Admin `gorm:"foreignKey:AdminID"` */

	// DiseaseID ทำหน้าที่เป็น FK
	DiseaseID uint
	Disease Disease `gorm:"foreignKey:DiseaseID"`

	// 1 Mealplan เป็นเจ้าของได้หลาย Mealday
	Mealdays []Mealday `gorm:"foreignKey:MealplanID"`
}